internal/services/oracle/oracle_database_system_versions_data_source.go

```diff
@@ -0,0 +1,180 @@
+			Optional:     true,
+			ValidateFunc: validation.StringInSlice(dbversions.PossibleValuesForBaseDbSystemShapes(), false),
+		},
+		"database_software_image_supported": {
+			Type:        pluginsdk.TypeBool,
+			Optional:    true,
+			Description: "Whether to filter versions that support database software images.",
+		},
+		"shape_family": {
+			Type:         pluginsdk.TypeString,
+			Optional:     true,
+			ValidateFunc: validation.StringInSlice(dbversions.PossibleValuesForShapeFamilyType(), false),
+			Description:  "The shape family of the DB System to filter versions by.",
+		},
+		"storage_management": {
+			Type:         pluginsdk.TypeString,
+			Optional:     true,
+			ValidateFunc: validation.StringInSlice(dbversions.PossibleValuesForStorageManagementType(), false),
+			Description:  "The storage management type to filter versions by.",
+		},
+		"upgrade_supported": {
+			Type:        pluginsdk.TypeBool,
+			Optional:    true,
+			Description: "Whether to filter versions that support upgrades.",
+		},
+	}
+}
+
+func (d DatabaseVersionsDataSource) Attributes() map[string]*pluginsdk.Schema {
+	return map[string]*pluginsdk.Schema{
+		"versions": {
+			Type:        pluginsdk.TypeList,
+			Computed:    true,
+			Description: "A list of available Oracle Database versions and their properties.",
+			Elem: &pluginsdk.Resource{
+				Schema: map[string]*pluginsdk.Schema{
+					"name": {
+						Type:     pluginsdk.TypeString,
+						Computed: true,
+					},
+					"latest_for_major_version": {
+						Type:     pluginsdk.TypeBool,
+						Computed: true,
+					},
+					"supports_pdb": {
+						Type:     pluginsdk.TypeBool,
+						Computed: true,
+					},
+					"version": {
+						Type:     pluginsdk.TypeString,
+						Computed: true,
+					},
+				},
+			},
+		},
+	}
+}
+
+func (d DatabaseVersionsDataSource) ModelObject() interface{} {
+	return &DatabaseVersionsModel{}
+}
+
+func (d DatabaseVersionsDataSource) ResourceType() string {
+	return "azurerm_oracle_database_system_versions"
+}
+
+func (d DatabaseVersionsDataSource) IDValidationFunc() pluginsdk.SchemaValidateFunc {
+	return dbversions.ValidateDbSystemDbVersionID
+}
+
+func (d DatabaseVersionsDataSource) Read() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 5 * time.Minute,
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			client := metadata.Client.Oracle.OracleClient.DbVersions
+			subscriptionId := metadata.Client.Account.SubscriptionId
+
+			var state DatabaseVersionsModel
+			if err := metadata.Decode(&state); err != nil {
+				return fmt.Errorf("decoding: %+v", err)
+			}
+
+			id := dbversions.NewLocationID(subscriptionId,
+				state.Location)
+
+			options := dbversions.DefaultListByLocationOperationOptions()
+
+			if state.ShapeFamily != "" || state.DatabaseSystemShape != "" {
+				log.Printf("[WARN] Db Versions data source: You can not use both shape_family and database_system_shape together for filtering versions.")
+			}
```

Comment:
consider using `ConflictsWith` in the schema definition of `Arguments()` function